CircleSpace/CircleSpaceServiceLib/Service/SqlCricleSpaceService.cs

```diff
@@ -0,0 +1,361 @@
+                });
+                db.SaveChanges();
+            }
+        }
+
+
+
+        public void AddPage(PageModel model)
+        {
+            using (var db = new CircleSpaceEntities())
+            {
+                var newPage = new Page()
+                {
+                    PageRoute = model.Route,
+                    ID = db.Pages.Max(p => p.ID) + 1,
+                    OwnerID = model.OwnerID,
+                    Header = model.Header,
+                    Body = model.Body,
+                    Footer = model.Footer
+                };
+            }
+        }
+        public void DeleteContributorFromPage(PageModel page, UserModel contributor)
+        {
+            using (var db = new CircleSpaceEntities())
+            {
+
+                var query = db.Pages.Where(p => p.ID == page.ID).First();
+                query.Users.Remove(db.Users.Where(u => u.ID == contributor.ID).First());
+                db.SaveChanges();
+
+            }
+        }
+
+        public void DeletePage(PageModel model)
+        {
+            using (var db = new CircleSpaceEntities())
+            {
+                var pageToDelete = db.Pages.Where(p => p.ID == model.ID).First();
+                db.Pages.Remove(pageToDelete);
+                db.SaveChanges();
+            }
+        }
+
+        public List<PageModel> GetContributorPages(UserModel model)
+        {
+            List<PageModel> list = new List<PageModel>();
+            using (var db = new CircleSpaceEntities())
+            {
+                var user = db.Users.Where(u => u.ID == model.ID).First();
+                var pages = user.Pages1.ToList();
+                pages.ForEach(page =>
+                {
+                    PageModel p = PageToPageModel(page);
+                    list.Add(p);
+                });
+            }
+
+            return list;
+        }
+
+        public List<LayoutModel> GetLayouts()
+        {
+            List<LayoutModel> list = new List<LayoutModel>();
+            using (var db = new CircleSpaceEntities())
+            {
+                var query = db.Layouts.Select(x => x);
+                list = LayoutsToListOfLayouts(query.ToList());
+            }
+
+            return list;
+        }
+
+        public LayoutModel GetLayoutWithID(int id)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.ID == id).First());
+            }
+            return l;
+        }
+
+        public LayoutModel GetLayoutWithOwner(UserModel model)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.OwnerID == model.ID).First());
+            }
+            return l;
+        }
+
+        public List<LayoutModel> GetLayoutsWithTag(params string[] tags)
```

Comment:
public List<LayoutModel> GetLayoutsWithTag(params string[] tags)
        {
            List<LayoutModel> layouts = null;

            using (var db = new CircleSpaceEntities())
            {
                List<Tag> tagsAsTagList = new List<Tag>();
                foreach (var tag in tags)
                {
                    tagsAsTagList.Add(new Tag() { Tag1 = tag });
                }
                var query = from layout in db.Layouts
                            where layout.Tags.SequenceEqual(tagsAsTagList, new TagNameComparer())
                            select layout;


            }
            return layouts;
        }